spec.html

```diff
@@ -28955,51 +28955,71 @@ <h1>
           <dd>It performs URI unescaping and decoding, preserving any escape sequences that correspond to Basic Latin characters in _preserveEscapeSet_.</dd>
         </dl>
         <emu-alg>
-          1. Let _strLen_ be the length of _string_.
+          1. Let _len_ be the length of _string_.
           1. Let _R_ be the empty String.
           1. Let _k_ be 0.
-          1. Repeat,
-            1. If _k_ = _strLen_, return _R_.
+          1. Repeat, while _k_ &lt; _len_,
             1. Let _C_ be the code unit at index _k_ within _string_.
-            1. If _C_ is not the code unit 0x0025 (PERCENT SIGN), then
-              1. Let _S_ be the String value containing only the code unit _C_.
-            1. Else,
-              1. Let _start_ be _k_.
-              1. If _k_ + 2 ≥ _strLen_, throw a *URIError* exception.
-              1. If the code units at index (_k_ + 1) and (_k_ + 2) within _string_ do not represent hexadecimal digits, throw a *URIError* exception.
-              1. Let _B_ be the 8-bit value represented by the two hexadecimal digits at index (_k_ + 1) and (_k_ + 2).
+            1. Let _S_ be _C_.
+            1. If _C_ is the code unit 0x0025 (PERCENT SIGN), then
+              1. If _k_ + 3 > _len_, throw a *URIError* exception.
+              1. Let _escape_ be the substring of _string_ from _k_ to _k_ + 3.
+              1. Let _B_ be ParseHexOctet(_string_, _k_ + 1).
+              1. If _B_ is not an integer, throw a *URIError* exception.
               1. Set _k_ to _k_ + 2.
               1. Let _n_ be the number of leading 1 bits in _B_.
               1. If _n_ = 0, then
-                1. Let _C_ be the code unit whose value is _B_.
-                1. If _C_ is not in _preserveEscapeSet_, then
-                  1. Let _S_ be the String value containing only the code unit _C_.
-                1. Else,
-                  1. Let _S_ be the substring of _string_ from _start_ to _k_ + 1.
+                1. Let _c_ be the code unit whose numeric value is _B_.
```

Comment:
There's already a `_C_`, so `_c_` is maybe not a good choice of name.

Reply:
I think it's fine, this alias does relate to the other one, and anyway is only used once and it's in the immediately following step. A later PR can improve _all_ the alias names in these operations.

Reply:
I also don't love this choice of names, but I agree the status quo is already bad.